Adding latest Deepseek 0528 model from FW, and Llama 4 Scout and Maverick

tawnymanticore opened the pull request:
## What does this PR do?

Adding new FW models

## Checklists

- [X] Tests have been run locally and passed
- [ ] New tests have been added to any work in /lib


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added support for three new AI models: DeepSeek R1 0528, Llama 4 Maverick Basic, and Llama 4 Scout Basic. These models are now available for selection and use within the application.
- **Tests**
  - Enhanced test utilities to provide more detailed information during structured output validation, improving test accuracy for models with reasoning capabilities.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tawnymanticore commented:
![Screenshot 2025-06-13 at 12 43 37 PM](https://github.com/user-attachments/assets/0bd38195-31af-4533-baa7-a851434d4f28)
no problem with JSON Schema for the llama 4 models. so that's all good.

Deepseek 0528 actually takes more than 10 minutes to run right now so our tests can't run on it... do I bump up that timeout or wait until FW gets more stable or figures out how to get it to run faster?

scosman commented on libs/core/kiln_ai/adapters/ml_model_list.py:
```diff
@@ -2298,6 +2301,47 @@ class KilnModel(BaseModel):
+        ],
+    ),
+    # Llama 4 Maverick Basic
+    KilnModel(
+        family=ModelFamily.llama,
+        name=ModelName.llama_4_maverick,
+        friendly_name="Llama 4 Maverick Basic",
+        providers=[
+            KilnModelProvider(
+                name=ModelProviderName.fireworks_ai,
+                model_id="accounts/fireworks/models/llama4-maverick-instruct-basic",
+                structured_output_mode=StructuredOutputMode.json_schema,
+            ),
+        ],
+    ),
```
@tawnymanticore can you take a look? I think the rabbit is right unless FW changed their APIs

scosman reviewed (changes requested):
We should add these for all providers we support. OpenRouter and Together probably have these. Grok may as well.

tawnymanticore commented:
> We should add these for all providers we support. OpenRouter and Together probably have these. Grok may as well.

**R1 0528**
-  Added OpenRouter entries for R1 0528
- Looks like Together went ahead and updated the "Deepseek R1" endpoint https://www.together.ai/models/deepseek-r1 with the 0528 endpoint and silently deprecated the old one, so we are good to go on Together as the existing entry covers it now.

**Scout**
- Added Openrouter, Together

**Maverick**
- Added Openrouter, Together
